cmd/main.go

```diff
@@ -222,17 +223,28 @@ func main() {
 
 	var objectHandler s3wrapper.API
 
+	switch Options.S3Mode {
+	case "aws":
+		objectHandler = s3wrapper.NewS3Client(&Options.S3Config, log, versionHandler, isoEditorFactory)
+		if objectHandler == nil {
+			log.Fatal("failed to create S3 client")
+		}
+	case "filesystem":
+		objectHandler = s3wrapper.NewFSClient(Options.JobConfig.WorkDir, log, versionHandler, isoEditorFactory)
+		if objectHandler == nil {
+			log.Fatal("failed to create S3 filesystem client")
+		}
+	default:
+		log.Fatalf("unsupported S3 mode: %s", Options.S3Mode)
+	}
```

Comment:
Can we put this logic somewhere in the s3wrapper package?

I think it would be a better API to do something like `s3wrapper.NewClient(...)` and have it return just an `s3wrapper.API` interface.

Reply:
Then the s3wrapper package config would track the type, rather than being in the main `Options` struct directly.

Reply:
> Can we put this logic somewhere in the s3wrapper package?

+1 to this idea! Perhaps, in the future, we cal also change the name of this package? Having it called `s3wrapper` and managing other backends can be misleading and confusing. Definitely not a priority now.

Reply:
I was kind of torn having all this logic directly in main. I'll work on it today along with the backwards compatibility for the env variables